### PR TITLE
[IMP] point_of_sale: extension point xml rendering

### PR DESCRIPTION
**Pull request by blaggacao**
Before, extending the xml rendering environment was a matter of overriding the
whole method.

This commit enables a clean extenstion point to alter the rendering dict in
the mean time till a thourough refactoring in the near future according to the
new js guidlines.

**Description of the issue/feature this PR addresses:**
- Missing (clean) extension point made alterations to the xml rendering context a real pain

**Current behavior before PR:**
- Unclean extension forces to rewrite the whole method.
- Maintainability of downstream code is suboptimal.

**Desired behavior after PR is merged:**
- Clean extension is possible
- (As if the this js code was already refactored according to the new style)

**Note:** Please do not close, even if not immediately accepted (`wishlist` tag). This PR is meant to be carried along and actively maintained, until this semantic will be completed this way or another.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

@xoe-labs

### CODE REVIEW

**Comment by blaggacao** [comment]
Note: Unfortunately, the diff is uglier than it needs to be...

**Comment by blaggacao** [comment]
@switch87 Could you have a kind look at this? The problem it solves is to be able to manipulate the receipt environment before passing it to the header and footer rendering, so that header and footer can make user of an extended environment.

This enables client-by-client customization in an _extended_ (legal) receipt environment.

**Comment by switch87** [comment]
Will do! :+1:

**Comment by switch87** [comment]
It will have to wait until next year.

**Comment by blaggacao** [comment]
Happy new year! I've no hurry, I'm maintaining a patched Odoo for the many reasons I sporadically try to push [one or the other PR](https://github.com/pulls?utf8=%E2%9C%93&q=is%3Aopen+is%3Apr+author%3Ablaggacao+archived%3Afalse+mentions%3Axoe-labs). If it will be included in master, I'm more than happy on the long run...

**Comment by blaggacao** [comment]
@switch87 Could you give this a go once again? It would be nice, could this be merged...

File: addons/point_of_sale/static/src/js/models.js [hunk 1]
```diff
@@ -2132,10 +2132,8 @@ exports.Order = Backbone.Model.extend({
             fiscal_position_id: this.fiscal_position ? this.fiscal_position.id : false
         };
     },
-    export_for_printing: function(){
+    _export_for_printing: function(){
```
Comment (switch87) [inline addons/point_of_sale/static/src/js/models.js]: rename function to what it actually does, now there are 2 functions, public and private, with the same name: for instance `_get_receipt_for_export`

Comment (blaggacao) [reply addons/point_of_sale/static/src/js/models.js]: @switch87 Done, thanks! force-pushed

**Comment by blaggacao** [comment]
@switch87 Shall we? :wink:

**Comment by switch87** [comment]
@pimodoo I think this would be an improvement. What do you think?

**Comment by pedrobaeza** [comment]
Is this still relevant?

**Pull request closed without merging**